frontend/src/lib/MetricsManager.ts

```diff
@@ -210,4 +217,27 @@ export class MetricsManager {
   private track(evName: string, data: Record<string, unknown>): void {
     analytics.track(evName, data)
   }
+
+  // Get the installation IDs from the session
+  private static getInstallationData(): Record<string, unknown> {
+    return {
+      machineIdV1: SessionInfo.current.installationIdV1,
+      machineIdV2: SessionInfo.current.installationIdV2,
+    }
+  }
+
+  // Use the tracking data injected by S4A if the app is hosted there
+  private static getHostTrackingData(): Record<string, unknown> {
+    if (isInChildFrame() && window.parent.streamlitShareMetadata) {
+      return pick(window.parent.streamlitShareMetadata, [
+        "hosted",
```

Comment:
Nit: can we rename this to "hostType" or "hostedAt" ?

Also, this can take values "s4a", "s4t", "localhost", right? But what if the app is in, say, Heroku? Should we have "other" as a valid string too? (Or replace "localhost" with "other" if Heroku shows as localhost today?)

Reply:
Sure, renaming to `hostedAt`

Currently we are only assigning `S4A` as a value to this and there are no restrictions on the list of values we can add.

So we can add `other` and `localhost` later on when we think of a way to detect and include this data.